Civi/Api4/Generic/Traits/DAOActionTrait.php

```diff
@@ -134,6 +134,11 @@ protected function writeObjects($items) {
         $item['contact_id'] = $entityId;
       }
 
+      // FIXME: Weird thing the Contribution BAO expects
+      if ($this->getEntityName() == 'Contribution') {
+        $item['skipCleanMoney'] = TRUE;
```

Comment:
@eileenmcnaughton *siren*

Reply:
Yeah - so I'm inclined to agree with this - I think it's still early enough in apiv4 we can make this change - which I would have liked to have made in v3 - ie the form layer should handle formatting

Reply:
I had to add this because without it the BAO emits

> Deprecated code path. Money should always be clean before it hits the BAO.

Reply:
Yep - apiv3 does the clean money in the api layer & then passes it through clean. But I think for v4 it's OK for the calling code to do the cleanup